iac-common/src/main/java/org/sonar/iac/common/extension/visitors/SyntaxHighlightingVisitor.java

```diff
@@ -56,4 +56,13 @@ protected void after(InputFileContext ctx, Tree root) {
   protected void highlight(HasTextRange range, TypeOfText typeOfText) {
     newHighlighting.highlight(range.textRange(), typeOfText);
   }
+
+  private void highlightComments(Tree tree) {
+    tree.comments().forEach(comment -> {
+      if(!commentTextRanges.contains(comment.textRange())) {
+        commentTextRanges.add(comment.textRange());
+        highlight(comment, COMMENT);
+      }
+    });
+  }
```

Comment:
Before my change, only a few `Tree` implementations returned some comments (because only a few classes implemented `HasComments` interface).
Now look for e.g. `AbstractDockerTreeImpl` - there is a new method that returns comments for all children. When traversing a tree the same comments will be returned. More highlighting for the same TextRange is not allowed by SonarQube. 

An alternative solution may be to return an empty list in `AbstractDockerTreeImpl.comments()`, but then the method needs to be implemented in each Docker Instruction - maintaining it will be a nightmare.